### DISCUSSION TITLE
This plugin is stable for Unity Desktop Build?

### DISCUSSION THREAD

**Discussion by princeSingh008** [WalletConnect/WalletConnectUnity#158] [Q&A] [answered]
Hi,
I am working on a unity desktop project which uses crypto currency for purchasing the in game things. I want to use this WalletConnect plugin for multiple wallet support. As I read that this plugin is not stable yet and also for WebGl build it is experimental. I just want to ask that is it stable for Unity Desktop application? Will it interrupt my development code in future releases?

**Comment by skibitsky** [comment, marked as answer]
Hi!

We don't anticipate any breaking changes moving forward. The API is expected to remain stable.

The packages should be stable on Windows and macOS. Linux isn’t officially supported, but it should work well there too. Regarding WebGL, it requires a third-party patcher. While it has proven stable in tests, we classify it as "experimental" due to its reliance on the patcher for proper operation.

**Comment by princeSingh008** [comment]
Ok, Thanks for your quick response. I am going to use this in my project and I will update you if I will face any issue.

**Comment by princeSingh008** [comment]
I want to use this plugin in my application as per my need with some modification (ETH chain and 3 wallets i.e. Metamask, Trust Wallet and Wallet Connect). If there is any documentation for the modification, kindly provide me else you can guide me for the same.

**Reply by skibitsky** [reply]
You can access the Unity documentation for WalletConnectModal [here](https://docs.walletconnect.com/advanced/walletconnectmodal/about?platform=unity).

It's compatible with all the wallets mentioned, as they all support the WalletConnect protocol.

To exclusively display Metamask and Trust Wallet options in the UI, simply add their IDs to the list of included wallets, as shown in the [documentation](https://docs.walletconnect.com/advanced/walletconnectmodal/options?platform=unity#includedwalletids). Additionally, you might want to customize the number of wallets shown in the Connect view (which is the initial screen with QR code and a selection of wallets) within Unity:

<img width=700 src="https://github.com/WalletConnect/WalletConnectUnity/assets/15127974/d7a32e9f-7658-4e6a-ad93-4ab9c6a764c8" />

---

There is also a [Sample](https://github.com/WalletConnect/WalletConnectUnity/tree/main/Packages/com.walletconnect.modal/Samples~/Modal%20Sample#readme) that you may find useful.

**Reply by princeSingh008** [reply]
Thank you soo much for your support. It worked fine for me as I wanted.
